packages/@aws-cdk/pipelines/lib/validation/shell-script-action.ts

```diff
@@ -72,6 +73,13 @@ export interface ShellScriptActionProps {
    * @default - No policy statements
    */
   readonly rolePolicyStatements?: iam.PolicyStatement[];
+
+  /**
+   * The VPC where to execute the specified script.
+   *
+   * @default - No VPC
+   */
+  readonly vpc?: ec2.IVpc;
```

Comment:
Probably also needs a way to do subnet selection then, no?

Reply:
Well, for the manually created `codebuild.PipelineProject` constructs, just specifying VPC is enough. As we can see from the test output in `validation.test.ts` the private subnets are properly selected: `VPCPrivateSubnet1Subnet8BCA10E0 `

Reply:
I *know* just specifying VPC can be enough, but it depends on your needs.

You're happy not allowing the selection today because *you* only need private subnets and the default happens to work for you. But tomorrow someone's going to come by complaining because they want to use public or isolated subnets and then we have to go through this again.

Please just add this property in as well and forward it:

https://github.com/aws/aws-cdk/blob/364543c9e7915783c27272f49326d8beada753ad/packages/%40aws-cdk/aws-codebuild/lib/project.ts#L487